app/models/google_calendar/adding.rb

```diff
@@ -0,0 +1,81 @@
+module GoogleCalendar
+  module Adding
+    AddEventInvalidParamsError = Class.new(StandardError)
+    AddEventInTimeSpanError = Class.new(StandardError)
+    AddEventInvalidRoom = Class.new(StandardError)
+
+    include Client
+
+    EVENT_SCHEMA = Dry::Validation.Schema do
+      required(:start).schema do
+        required(:date_time).filled
+      end
+
+      required(:end).schema do
+        required(:date_time).filled
+      end
+    end.freeze
+
+    def create(credentials, conference_room_id, raw_event_data = {})
+      event_data = build_event_data(raw_event_data, conference_room_id)
+      insert_event_and_return_result(credentials, event_data)
+    end
+
+    def insert_event_and_return_result(credentials, event_data)
+      raise_exception_if_occupied(credentials, event_data)
+      calendar_service(credentials).insert_event(
+        'primary',
+        Google::Apis::CalendarV3::Event.new(event_data)
+      )
+    end
+
+    def raise_exception_if_occupied(credentials, event_data)
+      events = events_in_span(credentials, event_data[:attendees].first,
+                              event_data[:start][:date_time], event_data[:end][:date_time])
+      if events && events.items.any?
```

Comment:
Can `events` be `nil`? Can we return an empty array instead of nil instead (in `events_in_span`)?